tests/unit_tests/test_ustar_cp/test_cpdAssignUStarTh20100901.py

```diff
@@ -0,0 +1,100 @@
+def test_cpdAssignUStarTh20100901_basic(matlab_engine, mock_stats):
+    # Convert mock stats to MATLAB structure
+    stats_matlab = matlab_engine.struct()
+    for field in ['mt', 'Cp', 'b1', 'c2', 'cib1', 'cic2', 'p']:
+        setattr(stats_matlab, field, matlab.double(getattr(mock_stats, field).tolist()))
+
+    fPlot = 0
+    cSiteYr = "TestSite_2024"
+
+    # Call MATLAB function
+    CpA, nA, tW, CpW, cMode, cFailure, fSelect, sSine, FracSig, FracModeD, FracSelect = matlab_engine.cpdAssignUStarTh20100901(
+        stats_matlab, fPlot, cSiteYr, nargout=11
+    )
+
+    # Assertions
+    assert isinstance(CpA, matlab.double), "CpA should be a MATLAB double array"
+    assert isinstance(nA, matlab.double), "nA should be a MATLAB double array"
+    assert isinstance(tW, matlab.double), "tW should be a MATLAB double array"
+    assert isinstance(CpW, matlab.double), "CpW should be a MATLAB double array"
+    assert isinstance(cMode, str), "cMode should be a string"
+    assert isinstance(cFailure, str), "cFailure should be a string"
+    assert isinstance(fSelect, matlab.logical), "fSelect should be a MATLAB logical array"
+    assert isinstance(sSine, matlab.double), "sSine should be a MATLAB double array"
+    assert isinstance(FracSig, float), "FracSig should be a float"
+    assert isinstance(FracModeD, float), "FracModeD should be a float"
+    assert isinstance(FracSelect, float), "FracSelect should be a float"
+
+def test_cpdAssignUStarTh20100901_plotting(matlab_engine, mock_stats):
+    # Test with plotting enabled
+    stats_matlab = matlab_engine.struct()
+    for field in ['mt', 'Cp', 'b1', 'c2', 'cib1', 'cic2', 'p']:
+        setattr(stats_matlab, field, matlab.double(getattr(mock_stats, field).tolist()))
+
+    fPlot = 1
+    cSiteYr = "TestSite_2024"
+
+    # Call MATLAB function
+    results = matlab_engine.cpdAssignUStarTh20100901(stats_matlab, fPlot, cSiteYr, nargout=11)
+
+    # Check if the function runs without errors when plotting is enabled
+    assert len(results) == 11, "Function should return 11 outputs even with plotting enabled"
+
+@pytest.mark.parametrize("invalid_input", [[], None, "invalid"])
+def test_cpdAssignUStarTh20100901_invalid_input(matlab_engine, invalid_input):
+    # Test with invalid input
+    invalid_stats = matlab.engine.struct()
+    invalid_stats.mt = matlab.double(invalid_input)  # Invalid input
+
+    fPlot = 0
+    cSiteYr = "TestSite_2024"
+
+    # Call MATLAB function
+    results = matlab_engine.cpdAssignUStarTh20100901(invalid_stats, fPlot, cSiteYr, nargout=11)
+
+    # Check if function handles invalid input gracefully
+    assert len(results[0]) == 0, "CpA should be empty for invalid input"
+    assert len(results[5]) > 0, "cFailure should contain an error message for invalid input"
+
+def test_cpdAssignUStarTh20100901_edge_cases(matlab_engine, mock_stats):
+    # Test edge cases (e.g., all significant change points, no significant change points)
+    edge_stats = matlab.engine.struct()
```

Comment:
This doesn't seem right to me. Shouldn't it be mathlab_engine.struct() since we need the runtime object to create us the struct?

Reply:
Fixed